Tools/build/check_warnings.py

```diff
@@ -214,11 +236,14 @@ def main(argv: list[str] | None = None) -> int:
         with Path(args.warning_ignore_file_path).open(
             encoding="UTF-8"
         ) as clean_files:
-            files_with_expected_warnings = {
-                file.strip()
+            # Files with expected warnings are stored as a set of tuples
```

Comment:
"set" has a specific meaning, we should use "list" instead.

Reply:
I had actually intended to make it a true set. All the elements in this list are unique

Reply:
Made it a set